node-graph/gcore/src/text.rs

```diff
@@ -29,3 +42,142 @@ impl From<TextAlign> for parley::Alignment {
+}
+
+impl fmt::Debug for Typography {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.debug_struct("Typography")
+			.field("font_family", &self.font_family)
+			.field("color", &self.color)
+			.field("stroke", &self.stroke)
+			.finish()
+	}
+}
+
+impl PartialEq for Typography {
+	fn eq(&self, _other: &Self) -> bool {
+		true
+	}
+}
+
+impl Hash for Typography {
+	fn hash<H: Hasher>(&self, state: &mut H) {
+		self.layout.len().hash(state);
+	}
+}
+
+impl Typography {
+	pub fn to_vector(&self) -> Table<Vector> {
+		Table::new()
+	}
+}
+
+#[derive(Clone)]
+pub struct NewFontCacheWrapper(pub Arc<Mutex<NewFontCache>>);
+
+impl fmt::Debug for NewFontCacheWrapper {
+	fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+		f.debug_struct("font cache").finish()
+	}
+}
+
+impl PartialEq for NewFontCacheWrapper {
+	fn eq(&self, _other: &Self) -> bool {
+		log::error!("Font cache should not be compared");
+		false
+	}
+}
+
+unsafe impl dyn_any::StaticType for NewFontCacheWrapper {
+	type Static = NewFontCacheWrapper;
+}
+
+pub struct NewFontCache {
+	pub font_context: parley::FontContext,
+	pub layout_context: parley::LayoutContext<()>,
+	pub font_mapping: HashMap<Font, (String, parley::fontique::FontInfo)>,
+	pub hash: u64,
+}
+
+impl NewFontCache {
+	pub fn new() -> Self {
+		let mut new = NewFontCache {
+			font_context: parley::FontContext::new(),
+			layout_context: parley::LayoutContext::new(),
+			font_mapping: HashMap::new(),
+			hash: 0,
+		};
+
+		let source_sans_font = Font::new(SOURCE_SANS_FONT_FAMILY.to_string(), SOURCE_SANS_FONT_STYLE.to_string());
+		new.register_font(source_sans_font, SOURCE_SANS_FONT_DATA.to_vec());
+		new
+	}
+
+	pub fn register_font(&mut self, font: Font, data: Vec<u8>) {
+		match self.font_mapping.entry(font) {
+			Entry::Occupied(occupied_entry) => {
+				log::error!("Trying to register font that already is added: {:?}", occupied_entry.key());
+			}
+			Entry::Vacant(vacant_entry) => {
+				let registered_font = self.font_context.collection.register_fonts(parley::fontique::Blob::from(data), None);
+				if registered_font.len() > 1 {
+					log::error!("Registered multiple fonts for {:?}. Only the first is accessible", vacant_entry.key());
+				};
+				match registered_font.into_iter().next() {
+					Some((family_id, font_info)) => {
+						let Some(family_name) = self.font_context.collection.family_name(family_id) else {
+							log::error!("Could not get family name for font: {:?}", vacant_entry.key());
+							return;
+						};
+						let Some(font_info) = font_info.into_iter().next() else {
+							log::error!("Could not get font info for font: {:?}", vacant_entry.key());
+							return;
+						};
+						let mut hasher = FxHasher::default(); // or FxHasher::new()
+						// Hash the Font for a unique id and add it to the cached hash
+						vacant_entry.key().hash(&mut hasher);
+						let hash_value = hasher.finish();
+						self.hash = self.hash.wrapping_add(hash_value);
```

Comment:
Instead of performing an addition here, you should hash the font data + the previous hash value. `FxHasher` is very fast but not particularly collision resistant and this could cause issues. There is also the question of why you use `FxHasher` anyway, it can't really be for perf reasons since you do also compute the hash of the same data using the default hasher. If you were concerned about performance we should only compute the hash once and reuse it for both use cases but I can guarantee you that the perf is going to be completely irrelevant in this case so idk.
If your intent was to make sure the insertion order does not affect the hash value, that is fine in principle but should an explaining commend and a justification why this does not create collisions